db/dbScript.sql

```diff
@@ -0,0 +1,393 @@
+  FOREIGN KEY (currency_id) REFERENCES Currency(id),
+  foreign key (city_id) references City(id),
+  UNIQUE (external_place_id)
+);
+create index if not exists idx_gym_external_place_id on Gym(external_place_id);
+create index if not exists idx_gym_lat_lon on Gym(latitude, longitude);
+create index if not exists idx_gym_price_changed_at on Gym(price_changed_at);
+create index if not exists idx_gym_owned_by on Gym(owned_by);
+create index if not exists idx_gym_city_id on Gym(city_id);
+
+CREATE TABLE if not exists Ownership
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),
+  requested_at TIMESTAMP NOT null check (requested_at <= NOW()), --time at which a gym account requested the ownership
+  responded_at TIMESTAMP check (responded_at > requested_at and responded_at <= NOW()), --time at which an admin made a decision regarding the ownership
+  status own_status NOT NULL, --status of the requested ownership: pending, approved, rejected
+  message text, --message the admin might write in order to explain his/her decision
+  responded_by uuid, --the admin account that made the decision
+  requested_by uuid NOT NULL, --the gym account that made the ownership request
+  gym_id uuid  NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (responded_by) REFERENCES Account(id),
+  FOREIGN KEY (requested_by) REFERENCES Account(id),
+  FOREIGN KEY (gym_id) REFERENCES Gym(id),
+  UNIQUE (gym_id, requested_by, status)
+);
+
+CREATE TABLE if not exists Request
+(
+  id uuid NOT null DEFAULT uuid_generate_v4(),
+  requested_at TIMESTAMP NOT null check (requested_at <= NOW()), --time of the request
+  origin_latitude FLOAT NOT null check (origin_latitude between -90 and 90), --latitude of the starting location
+  origin_longitude FLOAT NOT null check (origin_longitude between -180 and 180),
+  departure_time TIME, --preferred departure_time specified by user
+  weekday INT check (weekday between 0 and 6), --day of the week for which the user wants to receive the recommendations 
+  arrival_time TIME check (arrival_time > departure_time), --preferred arrival time specified by user
+  time_priority INT NOT null check (time_priority between 0 and 100), --the value of the travelling time slider on the frontend
+  tcost_priority INT NOT null check (tcost_priority between 0 and 100 and tcost_priority + time_priority = 100),
+  min_congestion_rating NUMERIC(4,2) NOT null check (min_congestion_rating between 1 and 5),
+  min_rating NUMERIC(4,2) NOT null check (min_rating between 1 and 5),
+  min_membership_price INT NOT null check (min_membership_price >= 0),
+  name VARCHAR(50),--each user can specify the name of the request 
+  user_id uuid NOT NULL,
+  PRIMARY KEY (id),
+  FOREIGN KEY (user_id) REFERENCES Account(id),
+  UNIQUE (user_id, name),
+  UNIQUE (user_id, requested_at)
+);
+create index if not exists idx_request_user_id on Request(user_id);
+create index if not exists idx_request_name on Request("name");
+
+CREATE TABLE if not exists Recommendation
```

Comment:
Could we revisit the use cases for the request-recommendation flow? Specifically, how are recommendations created—are they generated automatically by the system or manually by an admin user? 

If they are calculated automatically, we might not need this table to store them permanently, as it could be more efficient to recalculate them based on the request when needed.

Reply:
They are calculated automatically, but each user is able to browse through the history of his/her requests, so we need this table
In the Recommendation, we basically store the recommended gyms based on the chosen location of the user. So if we do not store anything, we would need to retrieve all the gyms once again and use Google Maps API once again, which is costly both timewise and financially